File: src/Entity/Offre.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Validator\Context\ExecutionContextInterface;
use App\Repository\OffreRepository;

#[ORM\Table(name: "offre")]
#[ORM\Entity(repositoryClass: OffreRepository::class)]
class Offre
{
    #[ORM\Column(name: "idOffre", type: "integer", nullable: false)]
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    private $idoffre;

    #[ORM\Column(name: "descriptionOffre", type: "string", length: 255, nullable: false)]
    private $descriptionoffre;

    #[ORM\Column(name: "nomOffre", type: "string", length: 255, nullable: false)]
    private $nomoffre;

    #[ORM\Column(name: "dateDebut", type: "date", nullable: false)]
    private $datedebut;

    #[ORM\Column(name: "dateFin", type: "date", nullable: false)]
    private $datefin;

    #[ORM\Column(name: "imageOffre", type: "string", length: 255, nullable: false)]
    private $imageoffre;

    #[ORM\Column(name: "reduction", type: "integer", nullable: false)]
    private $reduction;

    #[ORM\OneToMany(targetEntity: Produit::class, mappedBy: "offre")]

    private $produits;

    public function __construct()
    {
        $this->produits = new ArrayCollection();
    }

    public function getIdoffre(): ?int
    {
        return $this->idoffre;
    }

    public function getDescriptionoffre(): ?string
    {
        return $this->descriptionoffre;
    }

    public function setDescriptionoffre(string $descriptionoffre): static
    {
        $this->descriptionoffre = $descriptionoffre;
        return $this;
    }

    public function getNomoffre(): ?string
    {
        return $this->nomoffre;
    }

    public function setNomoffre(string $nomoffre): static
    {
        $this->nomoffre = $nomoffre;
        return $this;
    }

    public function getDatedebut(): ?\DateTimeInterface
    {
        return $this->datedebut;
    }

    public function setDatedebut(\DateTimeInterface $datedebut): static
    {
        $this->datedebut = $datedebut;
        return $this;
    }

    public function getDatefin(): ?\DateTimeInterface
    {
        return $this->datefin;
    }

    public function setDatefin(\DateTimeInterface $datefin): static
    {
        $this->datefin = $datefin;
        return $this;
    }

    public function getImageoffre(): ?string
    {
        return $this->imageoffre;
    }

    public function setImageoffre(string $imageoffre): static
    {
        $this->imageoffre = $imageoffre;
        return $this;
    }

    public function getReduction(): ?int
    {
        return $this->reduction;
    }

    public function setReduction(int $reduction): static
    {
        $this->reduction = $reduction;
        return $this;
    }

    /**
 * @return \Doctrine\Common\Collections\Collection|\App\Entity\Produit[]
     */
    public function getProduits(): Collection
    {
        return $this->produits;
    }

    public function addProduit(Produit $produit): self
    {
        if (!$this->produits->contains($produit)) {
            $this->produits[] = $produit;
            $produit->setOffre($this);
        }

        return $this;
    }

    public function removeProduit(Produit $produit): self
    {
        if ($this->produits->removeElement($produit)) {
            // Définir l'offre du produit à null
            $produit->setOffre(null);
        }

        return $this;
    }
}

File: src/main/java/entities/Offre.java
package entities;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Offre {
    private int idOffre;
    private String descriptionOffre;
    private String nomOffre;
    private Date dateDebut;
    private Date dateFin;
    private String imageOffre;
    private int reduction;
    List<Produit> produits;

    public Offre(int idOffre, String descriptionOffre, String nomOffre, Date dateDebut, Date dateFin, String imageOffre, int reduction) {
        this.idOffre = idOffre;
        this.descriptionOffre = descriptionOffre;
        this.nomOffre = nomOffre;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.imageOffre = imageOffre;
        this.reduction = reduction;
        this.produits = new ArrayList<>();
    }

    public Offre(int idOffre, String descriptionOffre, String nomOffre, Date dateDebut, Date dateFin) {
        this.idOffre = idOffre;
        this.descriptionOffre = descriptionOffre;
        this.nomOffre = nomOffre;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
    }


    public Offre(String nomOffre, String descriptionOffre,Date dateDebut, Date dateFin) {
        this.descriptionOffre = descriptionOffre;
        this.nomOffre = nomOffre;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
    }


    public Offre() {

    }

    public Offre( String descriptionOffre, String nomOffre, Date dateDebut, Date dateFin, int reduction) {
        this.descriptionOffre = descriptionOffre;
        this.nomOffre = nomOffre;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.reduction = reduction;
    }


    public int getIdOffre() {
        return idOffre;
    }

    public void setIdOffre(int idOffre) {
        this.idOffre = idOffre;
    }

    public String getDescriptionOffre() {
        return descriptionOffre;
    }

    public void setDescriptionOffre(String descriptionOffre) {
        this.descriptionOffre = descriptionOffre;
    }

    public String getNomOffre() {
        return nomOffre;
    }

    public void setNomOffre(String nomOffre) {
        this.nomOffre = nomOffre;
    }

    public Date getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(Date dateDebut) {
        this.dateDebut = dateDebut;
    }

    public Date getDateFin() {
        return dateFin;
    }

    public void setDateFin(Date dateFin) {
        this.dateFin = dateFin;
    }

    public String getImageOffre() {
        return imageOffre;
    }

    public void setImageOffre(String imageOffre) {
        this.imageOffre = imageOffre;
    }
    public int getReduction() {
        return reduction;
    }

    public void setReduction(int reduction) {
        this.reduction = reduction;
    }

    public List<Produit> getProduits() {
        return produits;
    }

    public void setProduits(List<Produit> produits) {
        this.produits = produits;
    }
    public void addProduit(Produit produit) {
        produits.add(produit);
        produit.setOffre(this); // Met à jour la référence à l'offre dans le produit
    }

    @Override
    public String toString() {
        return "Offre{" +
                "idOffre=" + idOffre +
                ", descriptionOffre='" + descriptionOffre + '\'' +
                ", nomOffre='" + nomOffre + '\'' +
                ", dateDebut=" + dateDebut +
                ", dateFin=" + dateFin +
                ", imageOffre='" + imageOffre + '\'' +
                ", reduction=" + reduction +
                ", produits=" + produits +
                '}';
    }


}